Add defaultSessionConfiguration to NSURLSessionConfiguration

sirnacnud opened the pull request:
- Removes NSURLSessionConfiguration.h
- Update NSURLSessionConfiguration defaultSessionConfiguration to property
- Implementation added in https://github.com/darlinghq/darling-cfnetwork/pull/2

CuriousTommy commented on include/Foundation/NSURLSessionConfiguration.h:
```diff
@@ -1,5 +1,7 @@
 #import <Foundation/NSObject.h>
 
 @interface NSURLSessionConfiguration : NSObject
-@end
 
+@property(class, readonly, strong) NSURLSessionConfiguration *defaultSessionConfiguration;
+
+@end
```
So it turns out that this header file (`NSURLSessionConfiguration.h`) actually doesn't exist in the macOS SDK.

The interface actually lives here:
https://github.com/darlinghq/darling-foundation/blob/c952468e11a5e3c41b1213d4f64649f0b09e8dd0/include/Foundation/NSURLSession.h#L148

I think it would be a good idea to replace `+ (NSURLSessionConfiguration *)defaultSessionConfiguration;` with `@property(class, readonly, strong) NSURLSessionConfiguration *defaultSessionConfiguration;`

sirnacnud replied:
Looks like this NSURLSessionConfiguration.h was not included in the build. I went ahead and removed it.

CuriousTommy reviewed (changes requested):
So the changes themselves are good, but I would personally redo the commits themselves to the following:

Commit 1:
```
Remove NSURLSessionConfiguration.h

(Add brief statement that explains about why we don't need this file. ex: The NSURLSessionConfiguration lives in NSURLSession.h, etc)
```

Commit 2:
```
Add defaultSessionConfiguration to NSURLSessionConfiguration
```

Other then that LGTM!

sirnacnud committed:
Remove NSURLSessionConfiguration.h

NSURLSessionConfiguration lives in NSURLSession.h and this duplicate defintion was not included in the build, so it can be removed.

sirnacnud committed:
Update NSURLSessionConfiguration defaultSessionConfiguration to property to match macOS defintion

CuriousTommy reviewed (approved):
Perfect, LGTM!
